File: bag-shop-web/src/app/models/order.ts
export type FulfillmentType =
  'DELIVERY' |
  'PICKUP';




export interface OrderItemRequest {

  productId: number;

  quantity: number;

}



export interface OrderRequest {

  shopSlug: string;

  customerName: string;

  customerPhone: string;

  customerEmail?: string;

  fulfillmentType: FulfillmentType;

  deliveryAddress?: string;

  deliveryCity?: string;

  items: OrderItemRequest[];
}




export interface OrderItemResponse {

  productId: number;

  productName: string;

  quantity: number;

  unitPrice: number;

  subtotal: number;

}


export interface OrderResponse {

  id: number;

  orderNumber: string;

  shopId: number;

  shopName: string;

  shopSlug: string;

  shopLogo?: string;


  status: string;

  fulfillmentType: FulfillmentType;

  subtotal: number;

  totalAmount: number;

  customerName: string;

  customerPhone: string;

  customerEmail?: string;


  deliveryAddress?: string;

  deliveryCity?: string;


  createdAt: string;


  items: OrderItemResponse[];

}

File: bag-shop-api/src/main/java/com/bag_shop_api/bag_shop_api/Entity/Order.java
package com.bag_shop_api.bag_shop_api.Entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import com.bag_shop_api.bag_shop_api.Enums.FulfillmentType;
import com.bag_shop_api.bag_shop_api.Enums.OrderStatus;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String orderNumber;
    
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
@JoinColumn(name = "shop_id", nullable = false)
private Shop shop;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FulfillmentType fulfillmentType;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal subtotal;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal totalAmount;

    // Customer information

    @Column(nullable = false)
    private String customerName;

    @Column(nullable = false)
    private String customerPhone;

    private String customerEmail;

    // Delivery information

    private String deliveryAddress;

    private String deliveryCity;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

}
